components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -412,9 +426,51 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
     }
 
     this._appendLine('Proceeding with project number: ' + projectNumber);
+    const token = await Gapi.getToken();
+    let readiness = false;
+    await request(
+      {
+        body: JSON.stringify(
+          {
+            Project: project,
+            ProjectNumber: projectNumber,
+            Token: token,
+          }
+        ),
+        headers: { 'content-type': 'application/json' },
+        method: 'PUT',
+        uri: this._configSpec.appAddress + '/initProject',
+      },
+      (error, response, body) => {
+        if (!error) {
+          try {
+            const msg = JSON.parse(response.body).Reply;
+            this._appendLine('project init succeeded: ' + msg);
+            readiness = true;
+          }
+          catch (e) {
+            this._appendLine('Backend returned non-json response: ');
+          }
+        } else {
+          this._appendLine('Backend error: ' + error);
+          return;
+        }
+      }
+    );
+
+    let initAttempts = 0;
+    const initTimeout = 2000;
+    do {
+      initAttempts++;
+      await wait(initTimeout);
+    } while (!readiness && initAttempts < 10);
```

Comment:
How does this work? `readiness` isn't being changed in the loop body. This will always either wait for 10*2000 ms or not wait at all.

Reply:
```readiness``` is changed in previous call back function, here the loop is to make sure call back updated the status.

Reply:
If you're `await`ing the request call (line 426), then the next code block will execute only when that call is finished and the callback is executed. It seems to me what you're trying to do here is run the request in parallel to this code block (starting line 456), which isn't how you should do this.

Reply:
In fact, I just noticed the `request` library isn't installed as a dependency, only it's typings. This should not be working.
You can just use the built-in `fetch` API. It's [supported](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility) by all major browsers.

Reply:
Which one is not working? This PR has been deployed to kubeflow.dev

Reply:
This code that calls `request(...` has not been deployed to kubeflow.dev. It's been a while since there's been a deployment as far as I'm aware.

The code I meant by not working is the call using `request`, that library isn't installed as part of the app, and this call will fail if the execution gets to it.

Reply:
We deployed this PR earlier on a new domain name. ```request``` seems works.

Reply:
- I see, that makes sense then, although we should explicitly install `request` as a dependency rather than rely on its existence by chance.
- Can you send me the new domain name?
- My original comment is still valid I believe. If you want to wait for the request to come back use a Promise (either using `request` or using the build-in `fetch`), rather than manually wait in a loop.

Reply:
new address is: kfdeploy.kubeflow.dev
merge PR to unblock @lluunn 
Will use promise in following PRs.